### PR TITLE
ddl: migrate part of ddl package code from Execute/ExecRestricted to safe API (1) (#22670)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #22670 to release-3.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/22930
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/22930:release-3.0-75f7485688ba
```

---

Signed-off-by: AilinKid <314806019@qq.com>

### What problem does this PR solve?

Issue Number: part close #pingcap/tidb-test#1152

### What is changed and how it works?

What's Changed:

Migrate part of code in ddl package from using Execute/ExecRestricted to safe API 

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- ddl: migrate part of ddl package code from Execute/ExecRestricted to safe API (1).

### CODE REVIEW

File: ddl/util/util.go [hunk 1]
```diff
@@ -145,20 +151,19 @@ func LoadDDLVars(ctx sessionctx.Context) error {
...
 
-const loadGlobalVarsSQL = "select HIGH_PRIORITY variable_name, variable_value from mysql.global_variables where variable_name in (%s)"
-
 // LoadGlobalVars loads global variable from mysql.global_variables.
 func LoadGlobalVars(ctx sessionctx.Context, varNames []string) error {
 	if sctx, ok := ctx.(sqlexec.RestrictedSQLExecutor); ok {
-		nameList := ""
+		var buf strings.Builder
+		sqlexec.MustFormatSQL(&buf, loadGlobalVars)
 		for i, name := range varNames {
```
Comment (xhebox) [inline ddl/util/util.go:159]: You can use `%?` to output `'xx','xx'` for `[]string` type.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

File: ddl/util/util.go (hunk 1 above)
Comment (AilinKid) [reply ddl/util/util.go]: addressed PTAL @xhebox

Comment (xhebox) [reply ddl/util/util.go]: Then these two invocation of `MustFormatSQL` can be combined into one.

**Review by tangenta — APPROVED** [review]

**Pull request merged**